.github/workflows/step_tests-pip.yml

```diff
@@ -85,6 +87,7 @@ jobs:
           python -m jupyterlab.browser_check
 
       - name: Test with pytest
+        continue-on-error: ${{ matrix.experimental }}
```

Comment:
Actually for this one I put the `continue-on-error` only if it's on `push` so that:
- when people look at the main branch they see green ticks
- when on PR we see what experimental jobs are failing and can navigate more easily
- use [workflow badge](https://github.com/LecrisUT/CMake-Template/blob/a4f73dbd53d8cb98e6325296c2905b2ed40dcaa3/README.md?plain=1#L3) to show overall pass/fail
- the pass job shows you when non-experimental jobs fail. Use that as a [required job](https://github.com/spglib/spglib/pull/358#issuecomment-1808113509) to keep trackof PR status instead